solenoid/src/compiler.rs

```diff
@@ -0,0 +1,1163 @@
+                self.push_label(name, builder);
+                let sp = self.build_sp(builder);
+                let x = self.build_peek(builder, sp, 2, "x");
+                let i = self.build_peek(builder, sp, 1, "x");
+                let sp = self.build_decr(builder, sp, 2);
+                // y = (x >> (248 - i * 8)) & 0xFF
+                let i = builder.build_left_shift(i, self.i256(3), "i");
+                let sub = builder.build_int_sub(self.i256(248), i, "sub");
+                let rr = builder.build_right_shift(x, sub, false, "rr");
+                let value = builder.build_and(rr, self.i256(0xFF), "ret").into();
+                self.build_push(builder, value, sp);
+            }
+            Instruction::Log(_) => {
+                error!("Event emission is unimpl: {:?}", instr);
+            }
+            Instruction::Stop => {
+                let name = "stop";
+                self.push_label(name, builder);
+                builder.build_return(None);
+            }
+            Instruction::SelfDestruct => {
+                error!("{:#?}", instr);
+                return None;
+            }
+            Instruction::CallDataLoad => {
+                let name = "calldataload";
+                self.push_label(name, builder);
+                let sp = self.build_sp(builder);
+                let (idx, sp) = self.build_pop(builder, sp);
+                let calldata = self.fun.unwrap().get_nth_param(0).unwrap().into_pointer_value();
+                let ptr = unsafe { builder.build_gep(calldata, &[idx], name)};
+                let ptr = builder.build_pointer_cast(ptr, self.i256_ty.ptr_type(AddressSpace::Generic), "ptr");
+                let value = builder.build_load(ptr, "value");
+                self.build_push(builder, value, sp);
+                let ptr_i8 = self.build_tos_ptr(builder, 1);
+                builder.build_call(self.swap_endianness(), &[ptr_i8.into()], "swap_endian");
+            }
+            Instruction::CallDataSize => {
+                let name = "calldatasize";
+                self.push_label(name, builder);
+                let sp = self.build_sp(builder);
+                let calldatasize = self.fun.unwrap().get_nth_param(1).unwrap().into_int_value();
+                let calldatasize = builder.build_int_cast(calldatasize, self.i256_ty, "calldatasize").into();
+                self.build_push(builder, calldatasize, sp);
+            }
+            Instruction::Invalid => {
+                let name = "invalid";
+                self.push_label(name, builder);
+                builder.build_unconditional_branch(self.errbb.unwrap());
+                self.pop_label();
+
+                if is_runtime {
+                    warn!("Invalid instruction encountered. Continuing compilation!");
+                    return Some(())
+                } else {
+                    warn!("Invalid instruction encountered. Halting compilation!");
+                    return None;

```

Comment:
This may be problematic if the constructor has complex branching logic. Should compile all the instructions.